pkg/reconciler/revision/cruds.go

```diff
@@ -109,3 +113,48 @@ func (c *Reconciler) createKPA(ctx context.Context, rev *v1alpha1.Revision) (*kp
 
 	return c.ServingClientSet.AutoscalingV1alpha1().PodAutoscalers(kpa.Namespace).Create(kpa)
 }
+
+func (c *Reconciler) checkAndUpdateKPA(ctx context.Context, rev *v1alpha1.Revision, have *kpav1alpha1.PodAutoscaler) (*kpav1alpha1.PodAutoscaler, changed, error) {
+	logger := logging.FromContext(ctx)
+	rawDesiredKPA := resources.MakeKPA(rev)
+
+	desiredKPA := have.DeepCopy()
+
+	desiredKPA.Spec = rawDesiredKPA.Spec
+	// Carry over new labels and annotations.
+	desiredKPA.Annotations = presources.UnionMaps(desiredKPA.Annotations, rawDesiredKPA.Annotations)
+	desiredKPA.Labels = presources.UnionMaps(desiredKPA.Labels, rawDesiredKPA.Labels)
+
+	haveUpdateContent := updateContent{
+		Annotations: have.Annotations,
+		Labels:      have.Labels,
+		Spec:        have.Spec,
+	}
+
+	desiredUpdateContent := updateContent{
+		Annotations: desiredKPA.Annotations,
+		Labels:      desiredKPA.Labels,
+		Spec:        desiredKPA.Spec,
+	}
+
+	diff, err := kmp.SafeDiff(desiredUpdateContent, haveUpdateContent)
+	if err != nil {
+		return nil, unchanged, fmt.Errorf("failed to diff kpa: %v", err)
+	}
+	if len(diff) == 0 {
```

Comment:
```suggestion
	if diff == "" {
```
Is the more standard way, I think.